routes/movies.js

```diff
@@ -0,0 +1,73 @@
+const router = require('express').Router();
+const Celebrity = require("../models/Celebrity");
+const Movie = require('../models/Movie');
+
+
+router.get('/movies', (req, res, next) => {
+  Movie.find()
+  .populate('cast')
+   .then(movies => {
+     res.render('movies/index', { moviesList: movies })
+   })
+   .catch(err => {
+    next(err);
+   })
+})
+
+router.get('/movies/newMovie', (req, res, next) => {
+  Celebrity.find()
+    .then(celebs => {
+      console.log(celebs)
+    res.render('movies/newMovie.hbs', { celebrities: celebs })
+    })
+    .catch(err => {
+    next(err);
+    })
+})
+
+router.post('/movies', (req, res, next) => {
+  const { title, genre, plot, cast } = req.body;
+  Movie.create({
+    title: title,
+    genre: genre,
+    plot: plot,
+    cast: cast
+  })
+    .then(movies => {
+      console.log(movies)
+      res.redirect(`/movies`)
+    })
+    .catch(err => {
+      next(err);
+    })
+})
+
+router.get('/movies/edit/:id', (req, res, next) => {
+  Movie.findById(req.params.id)
+  .then(movie => {
+    console.log(movie)
+    res.render('movies/edit', { movieData: movie })
+  })
+  .catch(err => {
+    next(err);
+  })
+})
```

Comment:
If you would like the user to be able to edit the cast as well, you can find all the celebrities from the database and pass it to res.render an object with two keys: `movieData` and `celebrityData`.

Reply:
Thanks for your feedback. I'll try to implement the iteration 12 as well :)